### PR TITLE
Campaign details charts for Spend, Impressions and Clicks

### PR DESCRIPTION
**Pull request by j6ll**
## Proposed Changes

This adds the new charts found in the PRD for improving campaign metrics (pedfti-7HS-p2)

This is the first of several PRs, treat this PR like a checkpoint and unless there changes that should block us for shipping, I'd like to follow up with suggestions in subsequent PRs so we can ship this improvement sooner.

Belo are the supplied designs
386186896-02aa0fa6-4529-4404-8f9a-f43f1305a9a0

There are some small deviations from the design for speed (highlighted red), we can iterate on that in subsequent PRs
- The tooltip is not implemented, I'm just using the chart default (see the chart in the video demo below)
- I'm not showing stats for the campaign duration (see how they grey out in the designs) I don't think this is practical for things like evergreen campaigns.  From testing I find the charts also look nicer with shorter date ranges, so I think we should skip this part
- There is no date filtering yet, it defaults to whole campaign

## Testing Instructions

- Open up several completed campaigns, you can use the calypso live link int he comments.  I suggest a variety - short campaigns, long campaigns cancelled campaigns etc.
- I've intentionally skipped adding a loading indicator. Since Jetpack sites can't access the new endpoint yet. So I just hide the data completely unless it exists for now
- Hover to show breakdowns for each day
- Toggle between clicks and impressions using the button above that graph
- Change the colour scheme for your user - the graph colours should update 

Note: I made some subtle styling tweaks since I recorded this, but is 99% the same

https://github.com/user-attachments/assets/fd748798-ab61-4ece-a16a-240cd01b3ee5

I have also displayed evergreen data in the following way (design only covered fixed length)

**Active**
Screenshot 2024-11-15 at 12 00 33

**Completed**
Screenshot 2024-11-15 at 12 01 02

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/lib/promote-post/index.ts [hunk 1]
```diff
@@ -328,10 +329,11 @@ export const requestDSPHandleErrors = async < T >(
 	siteId: number,
 	apiUri: string,
 	method: SupportedDSPMethods = 'GET',
-	body: Record< string, unknown > | undefined = undefined
+	body: Record< string, unknown > | undefined = undefined,
+	apiVersion = 1
```
Comment (therocket-gr) [inline client/lib/promote-post/index.ts]: i think it is cleared to use 
apiVersion: string = 'v1'

passing a number like 1.1 might have unexpected results

**Comment by therocket-gr** [comment]
i think we should tweak the styles here and maybe remove the border (or fix the padding)
it is sticking to the text and it doesn't look that good

Screenshot 2024-11-19 at 11 04 02 AM
Screenshot 2024-11-19 at 11 03 52 AM

maybe we should have some kind of highlight for the selected item in the drop-down

File: client/lib/promote-post/index.ts (hunk 1 above)
Comment (j6ll) [reply client/lib/promote-post/index.ts]: Ah I actually thought the opposite, I originally made it as a string and thought a number would be easier to validate?

I'm already using 1.1 in[ the code above](https://github.com/Automattic/wp-calypso/pull/96346/files#diff-48799417c950d435b194ceafd46e75b18f31ea45fe1d33ba1f2247aca343ca7cR179) and is working as expected.

But I'm not super opinionated, I can make it a string 👍

**Comment by j6ll** [comment]
> i think we should tweak the styles here and maybe remove the border (or fix the padding) it is sticking to the text and it doesn't look that good
> 
> Screenshot 2024-11-19 at 11 04 02 AM Screenshot 2024-11-19 at 11 03 52 AM
> maybe we should have some kind of highlight for the selected item in the drop-down

Thanks Vassilis, good shout, will fix 👍

File: client/lib/promote-post/index.ts (hunk 1 above)
Comment (therocket-gr) [reply client/lib/promote-post/index.ts]: i think it reads better also for developers 
having a `'v1.1'` is more clear than having a number like 1.1

Comment (therocket-gr) [reply client/lib/promote-post/index.ts]: for validation we could do something like that

`type SupportedApiVersions = 'v1' | 'v1.1';`

`apiVersion: SupportedApiVersions = 'v1'`

**Review by therocket-gr — APPROVED** [review]
works as described
code changes LGTM

**Review by therocket-gr — APPROVED** [review]

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17010095).

**Pull request merged**